utility/mm2_crawler

```diff
@@ -1003,22 +1042,29 @@ def per_host(session, host, options, config):
         # which is accessible via http or ftp
         successful_categories += 1
 
+        if options.canary:
+            continue
+
         trydirs = list(hc.category.directories)
-        # check the complete category in one go with rsync
-        try:
-            has_all_files = try_per_category(
-                session, trydirs, categoryUrl, host_category_dirs, hc,
-                host, categoryPrefixLen, options.timeout_minutes, config)
-        except TimeoutException:
-            # If the crawl of only one category fails, the host
-            # is completely marked as not being up to date.
-            raise
 
-        if type(has_all_files) == type(True):
-            # all files in this category are up to date, or not
-            # no further checks necessary
-            # do the next category
-            continue
+        # No rsync in canary mode, we only retrive a small subset of
+        # existing files
+        if not options.repodata:
+            # check the complete category in one go with rsync
+            try:
+                has_all_files = try_per_category(
+                    session, trydirs, categoryUrl, host_category_dirs, hc,
+                    host, categoryPrefixLen, options.timeout_minutes, config)
+            except TimeoutException:
+                # If the crawl of only one category fails, the host
+                # is completely marked as not being up to date.
+                raise
+
+            if type(has_all_files) == type(True):
```

Comment:
Interesting. This has always been there. The block has just been indented. The function returning has_all_files returns None, True, False. Looking at it I think it could be changed to only return True or False. I will add an additional commit to this PR to fix this.